lib/matplotlib/image.py

```diff
@@ -275,6 +275,13 @@ def __init__(self, ax,
 
         self._internal_update(kwargs)
 
+    def __str__(self):
+        try:
+            size = self.get_size()
+            return f"{type(self).__name__}(size={size!r})"
+        except RuntimeError:
```

Comment:
In what instances is this RuntimeError emitted? Perhaps worth a quick comment underneath?

Reply:
`get_size()` emit a RuntimeError if no data array is present. See some lines below.